src/main/java/pl/ergohestia/ehj1/ivesta/services/RouteService.java

```diff
@@ -1,22 +1,70 @@
 package pl.ergohestia.ehj1.ivesta.services;
 
-import pl.ergohestia.ehj1.ivesta.model.Driver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import pl.ergohestia.ehj1.ivesta.model.Route;
+import pl.ergohestia.ehj1.ivesta.model.TransportType;
+
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+public class RouteService implements Service<Route> {
+
+    // TODO remove
+    private static final Logger SYSOUT = LoggerFactory.getLogger("SYSOUT");
+
+    private final List<Route> routeList;
 
-public class RouteService implements Service<Route>  {
     public RouteService() {
+        this.routeList = new ArrayList<>();
+    }
+
+    public List<Route> getRoutesList() {
+        return routeList;
+    }
+
+    public void addRouteToList(Route route) {
+        routeList.add(route);
+    }
+
+    public Integer positiveIntegerValidator(Scanner scanner) {
+        Integer positiveInteger = 0;
+        boolean correctData = false;
+        do {
+            try {
+                positiveInteger = scanner.nextInt();
+                if (positiveInteger <= 0) {
+                    SYSOUT.info("Podaj liczbę całkowitą dodatnią");
+                } else {
+                    correctData = true;
```

Comment:
jeśli to jest w odrębnej metodzie, to ta zmienna `correctData` jest zbędna, bo wystarczy w tym miejscu zrobić `return`

Reply:
zmieniłem pętlę w metodzie z do while na while bez zmiennej correctData